File: app/src/main/java/io/githup/limvot/mangaapp/Manga.java
package io.githup.limvot.mangaapp;

import org.luaj.vm2.LuaTable;

/**
 * Created by nathan on 8/22/14.
 */
public class Manga {
    private int sourceNumber;
    private LuaTable backingTable;
    public Manga(int sourceNum, LuaTable tableIn) {
        sourceNumber = sourceNum;
        backingTable = tableIn;
    }
    public String toString() {
        return backingTable.get("title").toString();
    }
    public String getTitle() {
        return backingTable.get("title").toString();
    }
    public String getDescription() {
        String description =  backingTable.get("description").toString();
        if (description == "nil")
            return "No Description";
        return description;
    }
    public LuaTable getTable() {
        return backingTable;
    }
    public int getSourceNumber() {
        return sourceNumber;
    }
}

File: app/src/main/java/io/githup/limvot/mangaapp/Manga.scala
package io.githup.limvot.mangaapp;

import org.luaj.vm2.LuaTable;

class Manga(sourceNumber:Int, backingTable:LuaTable) {
  
    override def toString() : String = backingTable.get("title").toString()

    def getTitle() : String = backingTable.get("title").toString()

    def getDescription() : String = {
        val description = backingTable.get("description").toString()
        if (description == "nil")
            "No Description"
        else
            description
    }

    def getTable() : LuaTable = backingTable

    def getSourceNumber() : Int = sourceNumber
}
